module/VuFind/tests/unit-tests/src/VuFindTest/OAuth2/Repository/AbstractTokenRepositoryTestCase.php

```diff
@@ -172,57 +137,126 @@ protected function getMockUserTable(): User
     }
 
     /**
-     * Create AccessToken row
+     * Create User row
      *
      * @param array $data Row data
      *
-     * @return MockObject&AccessTokenRow
+     * @return MockObject&UserRow
      */
-    protected function createAccessTokenRow(array $data): AccessTokenRow
+    protected function createUserRow(array $data): UserRow
     {
-        $result = $this->getMockBuilder(AccessTokenRow::class)
+        $result = $this->getMockBuilder(UserRow::class)
             ->disableOriginalConstructor()
-            ->onlyMethods(['initialize', 'save'])
+            ->onlyMethods(['initialize'])
             ->getMock();
         $result->populate($data);
+        return $result;
+    }
 
-        $save = function () use ($result) {
-            $data = $result->toArray();
-            foreach ($this->accessTokenTable as &$row) {
-                if (
-                    $data['id'] === $row['id']
-                    && $data['type'] === $row['type']
-                ) {
-                    $row = $data;
-                    return 1;
-                }
-            }
-            $this->accessTokenTable[] = $data;
-            return 1;
-        };
+    /**
+     * Mock entity manager.
+     *
+     * @return MockObject
+     */
+    protected function getEntityManager()
+    {
+        $entityManager = $this->getMockBuilder(\Doctrine\ORM\EntityManager::class)
+            ->disableOriginalConstructor()
+            ->onlyMethods(['createQuery','persist','flush'])
+            ->getMock();
+        $query = $this->createMock(\Doctrine\ORM\Query::class);
+        $entityManager->expects($this->any())->method('createQuery')->willReturn($query);
+        $entityManager->expects($this->any())->method('persist');
+        $entityManager->expects($this->any())->method('flush');
+        return $entityManager;
+    }
 
-        $result->expects($this->any())
-            ->method('save')
-            ->willReturnCallback($save);
+    /**
+     * Mock entity plugin manager.
+     *
+     * @param bool $setExpectation Flag to set the method expectations.
+     *
+     * @return MockObject
+     */
+    protected function getPluginManager($setExpectation = false)
+    {
+        $pluginManager = $this->getMockBuilder(
```

Comment:
You could simplify this to `$pluginManager = $this->createMock(\VuFind\Db\Entity\PluginManager::class);` since you're not using any non-default behaviors of the mock builder.